### ISSUE TITLE
Cut off deeper searches if shallower checkmate already found

### ISSUE THREAD

**Issue by MarkZH** [MarkZH/Genetic_Chess#32]
Something like:
```
Game_Tree_Result game_tree_search(...)
{
    // ...
    auto best_result = ...; // first move
    if(alpha.corrected_score(perspective) == Math::win_score && alpha.depth <= depth)
    {
        return alpha;
    }
    // ...
}

**Comment by MarkZH** [comment]
Won't fix.

1. This is how alpha-beta pruning works in general.
2. The search can't be cut off since the opponent may not allow the shallower checkmate to be reached.

**Comment by MarkZH** [comment]
Duplicate of fixed issue #37
